ecodse_funtime_alpha/train.py

```diff
@@ -16,24 +19,49 @@ def train_loop(dataset, model, optimizer):
         print(loss)
 
 
-def fit_loop(dataset, model, optimizer):
-    dataset = dataset.shuffle(12)
-    dataset = dataset.repeat(20)
-    dataset = dataset.batch(4)
+def fit_loop(dataset, model, optimizer, nepoch, batchsize):
+    nstep = len(list(dataset)) // batchsize
+    dataset = dataset.shuffle(12)  # 12 = buffer size
+    dataset = dataset.repeat(nepoch)
+    dataset = dataset.batch(batchsize)
     model.compile(optimizer=optimizer,
                   loss=tf.keras.losses.binary_crossentropy,
                   metrics=["accuracy"])
-    model.fit(dataset, epochs=20, steps_per_epoch=2)
+    model.fit(dataset, epochs=nepoch, steps_per_epoch=nstep)
+
+
+def get_args(args):
+    argparser = argparse.ArgumentParser()
+    argparser.add_argument('--imagepath', default='../../rainforest/fixed-train-jpg/',
+                           help='path to image directory')
+    argparser.add_argument('--labelpath', default='../../rainforest/train_v3.csv',
+                           help='path to csv file for labels')
+    argparser.add_argument('--seed', default=-1, type=int,
+                           help='Set random seed to this number (if >0)')
+    argparser.add_argument('--kernels', default=1, type=int,
+                           help='Number of kernels in the CNN')
+    argparser.add_argument('--ksize', default=1, type=int,
+                           help='Size of kernels in CNN')
+    argparser.add_argument('--lr', default=0.1, type=float,
+                           help='Learning rate (default=0.1')
+    argparser.add_argument('--nepoch', default=1, type=int,
+                           help='Number of epoch for training')
+    argparser.add_argument('--batchsize', default=32, type=int,
+                           help='batch size (default=32)')
+    args = argparser.parse_args(args)
+    return args
 
 
 if __name__ == "__main__":
-    image_dir = '/project/def-bengioy/hrb/rainforest/fixed-train-jpg/'
-    labels_csv = '/project/def-bengioy/hrb/rainforest/train_v2.csv'
-    dataset = data.get_dataset(image_dir, labels_csv)
+    tf.enable_eager_execution()
+    args = get_args(sys.argv[1:])
+    if args.seed > 0:
+        tf.random.set_seed(args.seed)
+    dataset, lendataset = data.get_dataset(args.imagepath, args.labelpath)
```

Comment:
I don't think that adding a new return to get_dataset in necessary in this PR for argparsr.
Please move this to another PR.

Reply:
moved to another branch.

Reply:
You forgot this one.

Reply:
You forgot to remove `lendataset` here